src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -45,13 +46,15 @@
     private $router;
     private $identifiersExtractor;
     private $identifierDenormalizer;
+    private $readListener;
 
-    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ItemDataProviderInterface $itemDataProvider, RouteNameResolverInterface $routeNameResolver, RouterInterface $router, PropertyAccessorInterface $propertyAccessor = null, IdentifiersExtractorInterface $identifiersExtractor = null, ChainIdentifierDenormalizer $identifierDenormalizer = null)
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ItemDataProviderInterface $itemDataProvider = null, RouteNameResolverInterface $routeNameResolver, RouterInterface $router, PropertyAccessorInterface $propertyAccessor = null, IdentifiersExtractorInterface $identifiersExtractor = null, ChainIdentifierDenormalizer $identifierDenormalizer = null, ReadListener $readListener = null)
```

Comment:
It introduces a dependency to a Symfony full stack related class in the low level infrastructure. I don't think it's a good idea, it will be very painful to support other frameworks (or PSR-7) in the future.

Reply:
Okay WDYT about this:

1. Move private methods to a trait
2. Introduce a new class (`OperationDataProvider` that uses the trait)
3. Inject the `OperationDataProvider` in the IriConverter so that it can call item and subresource operations

This way: 
- no changes in read listener
- de-duplicate code in IriConverter
- improves future updates in the IriConverter concerning data retrieval (IriToItem)